pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/ISOMetadataExtractorConfigurationFactory.java

```diff
@@ -143,52 +145,227 @@ public DocumentProcessingConfiguration createMetadataDocumentProcessingConfigura
+				"(//gmd:identificationInfo/*/gmd:graphicOverview/*/gmd:fileName/gco:CharacterString/text())"));
+
+		// rerource id
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.RESOURCE_ID,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:citation/*/gmd:identifier/*/gmd:code/gco:CharacterString/text())"));
+
+
+
+		// Lineage / source of the metadata
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LINEAGE,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:dataQualityInfo/*/gmd:lineage/*/gmd:statement/gco:CharacterString/text())"));
+
+
+		/*
+			Dates
+		*/
 
 		// Datestamp
 		extractors.add(createXPathExtractor(
 				ResultMetadataFields.DATESTAMP,
 				FieldExtractorType.FIRST_MATCHING_VALUE,
 				"//gmd:MD_Metadata/gmd:dateStamp/*/text()"));
 
+		// Published date
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.DATE_PUBLISHED,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:citation/*/gmd:date/*/gmd:date/gco:Date/text())"));
+
+		// Get all identification info dates
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.DATE_IDENTIFICATION_INFO,
+				new DateXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:citation/*/gmd:date)"));
+
+		// maintenance frequency
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.MAINENANCE_FREQUENCY,
+				new CodeListValueXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:resourceMaintenance/*/gmd:maintenanceAndUpdateFrequency/*[@codeListValue])"));
+
+		/*
+			Organisations
+		 */
 
 		// Organisation names + roles
 		extractors.add(createXPathExtractor(
 				ResultMetadataFields.ORGANISATIONS,
 				new ResponsiblePartyXmlCustomExtractor(),
 				"//gmd:MD_Metadata/gmd:identificationInfo/*/gmd:pointOfContact/gmd:CI_ResponsibleParty"));
 
+
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_METADATA,
+				new ResponsiblePartyXmlCustomExtractor(),
+				"//gmd:MD_Metadata/gmd:contact/gmd:CI_ResponsibleParty"));
+
+		/*
+			Language data
+		 */
+		// Metadata language
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LANGUAGE_METADATA,
+				new LanguageXmlCustomExtractor(),
+				"(/csw:GetRecordByIdResponse/*/gmd:language)"));
+
+		//Resource language
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.LANGUAGE_RESOURCE,
+				new LanguageXmlCustomExtractor(),
+				"(//gmd:identificationInfo/*/gmd:language)"));
+
+
+
+		/*
+		 Constraints
+		*/
+		// use limitation
+		extractors.add(createXPathExtractor(
+				ResultMetadataFields.CONSTRAINT_USE_LIMITATION,
+				FieldExtractorType.FIRST_MATCHING_VALUE,
+				"(//gmd:identificationInfo/*/gmd:resourceConstraints/*/gmd:useLimitation/gco:CharacterString/text())"));
```

Comment:
Not all texts are as gco:CharacterString, that should be switched to a "*" in the matcher

Reply:
Removed most occurrences - xpath for  `ResultMetadataFields.KEYWORDS_INSPIRE` seems to need these for a correct result (?). Tests fail if changed to a "*" matcher

Reply:
Inspire keywords seem to confuse the * matcher as the example documents contain alternate language versions. I switched gco:CharacterString to gco:* for Inspire